js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+                //     }
+                //
+                ordersOfTrigger = this.safeValue (response, 'data');
+            }
+            const merged = this.arrayConcat (ordersOfTrigger, ordersOfRegular);
+            return this.parseOrders (merged, market, since, limit, params);
+        }
+    }
+
+    async fetchOrdersByIds (ids, symbol = undefined, params = {}) {
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchOrdersByIds', undefined, params);
+        await this.loadMarkets ();
+        const request = {};
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['symbol'] = market['id'];
+        }
+        if (marketType === 'spot' || (market !== undefined && market['spot'])) {
+            throw new BadRequest (this.id + ' fetchOrdersByIds() is not supported for ' + marketType);
+        } else {
+            request['order_ids'] = ids.join (',');
+            const response = await this.contractPrivateGetOrderBatchQuery (this.extend (request, query));
+            //
+            //     {
+            //         "success": true,
+            //         "code": "0",
+            //         "data": [
+            //             {
+            //                 "orderId": "265230764677709315",
+            //                 "symbol": "STEPN_USDT",
+            //                 "positionId": "0",
+            //                 "price": "2.1",
+            //                 "vol": "102",
+            //                 "leverage": "20",
+            //                 "side": "1",
+            //                 "category": "1",
+            //                 "orderType": "1",
+            //                 "dealAvgPrice": "0",
+            //                 "dealVol": "0",
+            //                 "orderMargin": "10.96704",
+            //                 "takerFee": "0",
+            //                 "makerFee": "0",
+            //                 "profit": "0",
+            //                 "feeCurrency": "USDT",
+            //                 "openType": "1",
+            //                 "state": "2",
+            //                 "externalOid": "_m_7e42f8df6b324c869e4e200397e2b00f",
+            //                 "errorCode": "0",
+            //                 "usedMargin": "0",
+            //                 "createTime": "1648906342000",
+            //                 "updateTime": "1648906342000",
+            //                 "positionMode": "1"
+            //             }
+            //         ]
+            //     }
+            //
+            const data = this.safeValue (response, 'data');
+            return this.parseOrders (data, market);
+        }
+    }
+
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchOpenOrders', undefined, params);
```

Comment:
When calling `fetchOpenOrders` for the `swap` type ends up calling `fetchOrdersByState` and the latter calls `fetchOrders`.

The problem is that we're losing the marketType in the middle of this chain and `fetchOrder` assumes the `spot` market resulting in an error

As you can see in the image on line 1940 we omit it from params and then on line 1952 we call `fetchOrders` but `query` does not have the type anymore
![image](https://user-images.githubusercontent.com/43336371/164739951-5b1b2d05-9b9b-4653-a1bb-e6e183824a8f.png)

Reply:
haven't revised - left for you 👍